internal/types/subtypes/interceptor.go

```diff
@@ -297,8 +298,16 @@ func AttributeTransformerInterceptor(_ context.Context) grpc.UnaryServerIntercep
 	return func(interceptorCtx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		if reqMsg, ok := req.(proto.Message); ok {
 			if err := transformRequest(reqMsg); err != nil {
-				return nil, handlers.InvalidArgumentErrorf("Error in provided request.",
-					map[string]string{"attributes": "Attribute fields do not match the expected format."})
+				fieldErrs := map[string]string{
+					"attributes": "Attribute fields do not match the expected format.",
+				}
+
+				var invalidArgumentErr *InvalidArgumentError
+				if errors.As(err, &invalidArgumentErr) {
+					fieldErrs[invalidArgumentErr.FieldName] = invalidArgumentErr.Msg
+				}
```

Comment:
This error shouldn't add a new field, the error is still in the `attributes` field. I think in this case it might just override what's in the `attributes` key.

Reply:
this makes sense, thanks for the feedback